packages/p2p-v2/src/hooks/useQueryString.ts

```diff
@@ -0,0 +1,109 @@
+import { useEffect, useRef, useState } from 'react';
+import { useHistory } from 'react-router-dom';
+import { useEventListener } from 'usehooks-ts';
+
+declare global {
+    interface WindowEventMap {
+        queryChange: CustomEvent;
+    }
+}
+
+type TSetQueryStringOptions = {
+    clearAllQueries?: boolean;
+};
+
+type TQueryString = {
+    advertId?: string;
+    form?: string;
+    modal?: string;
+    paymentMethodId?: string;
+    tab?: string;
+};
+
+function mergeParams(params1: URLSearchParams, params2: URLSearchParams) {
+    const payload1 = paramsToPayload(params1);
+    const payload2 = paramsToPayload(params2);
+    const formattedParams = payloadToParams({
+        ...payload1,
+        ...payload2,
+    });
+
+    return formattedParams;
+}
+
+const payloadToParams = (payload: TQueryString) => new URLSearchParams(Object.entries(payload));
+const paramsToPayload = (params: URLSearchParams) => Object.fromEntries(params.entries());
+const queryEvent = new Event('queryChange');
+
+// TODO: Add validations for query strings once My Profile page is merged
```

Comment:
if not needed please remove these commented lines.

Reply:
I've added these for myself as reminder to add configurations for the query strings later